.github/actions/k-bench/evaluate/graph.py

```diff
@@ -0,0 +1,181 @@
+"""Generate graphs comparing K-Bench benchmarks across cloud providers and Constellation."""
+import json
+import os
+from collections import defaultdict
+
+import numpy as np
+from matplotlib import pyplot as plt
+
+SUBJECTS = [
+    "constellation-gcp",
+    "GKE",
+    "constellation-azure",
+    "AKS",
+]
+
+BAR_COLORS = ["#90FF99", '#929292', "#8B04DD", '#000000']
+
+# Rotate bar labels by X degrees
+LABEL_ROTATE_BY = 30
+LABEL_FONTSIZE = 9
+
+# Some lookup dictionaries for x axis
+api_suffix = 'ms'
+pod_key2header = {
+    'pod_create':   'Pod Create',
+    'pod_list':     'Pod List',
+    'pod_get':      'Pod Get',
+    'pod_update':   'Pod Update',
+    'pod_delete':   'Pod Delete',
+}
+svc_key2header = {
+    'svc_create':   'Service Create',
+    'svc_list':     'Service List',
+    'svc_update':   'Service Update',
+    'svc_delete':   'Service Delete',
+    'svc_get':      'Service Get',
+}
+depl_key2header = {
+    'depl_create':  'Deployment Create',
+    'depl_list':    'Deployment List',
+    'depl_update':  'Deployment Update',
+    'depl_scale':   'Deployment Scale',
+    'depl_delete':  'Deployment Delete',
+}
+
+fio_suffix = 'MiB/s'
+fio_key2header = {
+    'fio_root_async_R70W30_R':   'async_R70W30 mix,\n seq. reads',
+    'fio_root_async_R70W30_W':   'async_R70W30 mix,\n seq. writes',
+    'fio_root_async_R100W0_R':   'async_R100W0 mix,\n seq. reads',
+    'fio_root_async_R0W100_W':   'async_R0W100 mix,\n seq. writes',
+}
+
+net_suffix = 'Mbit/s'
+net_key2header = {
+    'net_internode_snd':            f'iperf internode \n send ({net_suffix})',
+    'net_intranode_snd':            f'iperf intranode \n send ({net_suffix})',
+}
+
+
+def configure() -> str:
+    """Read the benchmark data paths.
+
+    Expects ENV vars (required):
+    - BDIR=benchmarks
+
+    Raises TypeError if at least one of them is missing.
+
+    Returns: out_dir
+    """
+    out_dir = os.environ.get('BDIR', None)
+    if not out_dir:
+        raise TypeError(
+            'ENV variables BDIR is required.')
+    return out_dir
+
+
+def bar_chart(data, headers, title='', suffix='', val_label=True, y_log=False):
+    """Draws a bar chart with multiple bars per data point.
+
+    Args:
+        data (list): List of lists of value points.
+        headers (list): List of headers (x-axis).
```

Comment:
```suggestion
        data (dict[str, list]): List of lists of value points.
        headers (dict[str, list]): List of headers (x-axis).
```

Both are not lists.

Reply:
Headers should be a list right? It comes from ` dict.values()`.

Reply:
`dict.values()` doesn't directly return a list. You get some weird [view object](https://docs.python.org/3.11/library/stdtypes.html#dictionary-view-objects) that can also update dynamically.

For example, if you do this:
```python
mydict = {"first": True}
values = mydict.values()
print(values)
mydict["second"] = False
print(values)
```

You get:
```python
dict_values([True])
dict_values([True, False])
```

So my suggestion here technically also wasn't correct.

But if you want to return a list you can just cast it to a list with `list(mydict.values())`. That should be correct then.